gcc/rust/backend/rust-compile-context.h

```diff
@@ -341,6 +353,9 @@ class Context
   std::vector<::Bvariable *> var_decls;
   std::vector<tree> const_decls;
   std::vector<tree> func_decls;
+
+  // Nonzero iff we are currently compiling something inside a constant context.
+  unsigned int const_context = 0;
```

Comment:
This is great we need to port over from CPP DECL_DECLARED_CONSTEXPR_P which sets a lang_flag bit on all relevant tree's. We might call it the same name but when we are compiling any constants we could set this as well. thanks for doing it